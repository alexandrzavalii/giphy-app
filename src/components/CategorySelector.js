import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.section`
    padding: 1em;
    display: flex;
    justify-content: center;
`;

const Button = styled.button`
    background: ${props => props.selected ? props.theme.base : "initial"};
    color: ${props => props.theme.light};
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    border: 2px solid ${props => props.theme.base};
    border-radius: 3px;
        &:hover {
            cursor: pointer;
        }
`;

const propTypes = {
    activeCategory: PropTypes.string.isRequired,
    availableCategories: PropTypes.array.isRequired,
    handleClick: PropTypes.func.isRequired
}


export const CategorySelector = ({ activeCategory, availableCategories, handleClick }) => {
    let categories = availableCategories.length > 0 ?
        availableCategories.map(category =>
            <Button key={category}
                selected={category === activeCategory}
                onClick={() => handleClick(category)}> {category}</Button>)
        : null;
    return <Wrapper>{categories}</Wrapper>;
}

CategorySelector.propTypes = propTypes;