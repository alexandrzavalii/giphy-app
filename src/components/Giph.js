import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Gif = styled.img`
  width: ${props => props.selected ? '80%' : '100%'};
  height: auto;
  margin: ${props => props.selected ? '10%' : 'initial'};
`;

const propTypes = {
    giphSrc: PropTypes.string.isRequired,
    giphSlug: PropTypes.string.isRequired,
    selected: PropTypes.bool.isRequired,
    id: PropTypes.string.isRequired,
    handleGiphInteraction: PropTypes.func.isRequired
}

export class Giph extends React.PureComponent {
    constructor(props){
        super(props);
        this.selectGiph = this.selectGiph.bind(this);
    }
    
    selectGiph() {
        this.props.handleGiphInteraction(this.props.id);
    }

    render() {
        return <Gif selected={this.props.selected} src={this.props.giphSrc} alt={this.props.giphSlug} onMouseOver={this.selectGiph} />
    }
}

Giph.propTypes = propTypes;