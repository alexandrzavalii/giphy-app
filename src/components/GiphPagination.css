.pagination {
    display: inline-block;
}

.pagination a {
    color: black;
    float: left;
    padding: 8px 10px;
    text-decoration: none;
}

.pagination a.active {
    background-color: #4CAF50;
    color: white;
    border-radius: 5px;
}
.pagination a.disabled {
    display: none;
}
.pagination a:hover:not(.active) {
    background-color: #ddd;
    border-radius: 5px;
    cursor: pointer;
}