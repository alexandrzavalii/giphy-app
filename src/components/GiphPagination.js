import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { GIFS_PER_PAGE } from "../settings";

import styled from 'styled-components';

// Styling
const Wrapper = styled.section`
    display: flex;
    justify-content: center;
    margin: 20px 0;
`;

const PageLink = styled.a`
    color: black;
    float: left;
    padding: 8px 10px;
    text-decoration: none;
    display: ${props => props.disabled ? 'none' : 'initial'};
    background-color: ${props => props.active ? 'palevioletred' : 'white'};
    color: ${props => props.active ? 'white' : 'black'};
    border-radius: 3px;
    &:hover {
        cursor: pointer;
    }
`;

// PropTypes
const propTypes = {
    totalItems: PropTypes.number.isRequired,
    handlePageClick: PropTypes.func.isRequired
}
const defaultProps = {
    initialPage: 1
}

// Component
export class GiphPagination extends Component {
    constructor(props) {
        super(props);
        this.state = {
            pager: {}
        }
    }

    componentWillMount() {
        if (this.props.totalItems) {
            console.log("WILL MOUNT");
            this.setPage(this.props.initialPage);
        }
    }

    componentDidUpdate(prevProps, prevState) {
        // reset page if items array has changed
        if (this.props.totalItems !== prevProps.totalItems) {
            this.setPage(this.props.initialPage);
        }
    }


    setPage(page) {
        console.log("PAGE",page);
        console.log("PROPS", this.props.initialPage);
        if (page !== this.state.pager.currentPage) {
            var items = this.props.totalItems;
            var pager = this.state.pager;

            if (page < 1 || page > pager.totalPages) {
                return;
            }

            // get new pager object for specified page
            pager = this.getPager(items, page);
            this.setState({ pager: pager });

            // call change page function in parent component
            this.props.handlePageClick((page - 1) * GIFS_PER_PAGE);
        }
    }

    getPager(totalItems, currentPage) {
        // default to first page
        currentPage = currentPage || 1;

        // default page size is 10
        let pageSize = GIFS_PER_PAGE;

        // calculate total pages
        var totalPages = Math.ceil(totalItems / pageSize);
        var startPage, endPage;
        if (totalPages <= 10) {
            // less than 10 total pages so show all
            startPage = 1;
            endPage = totalPages;
        } else {
            // more than 10 total pages so calculate start and end pages
            if (currentPage <= 6) {
                startPage = 1;
                endPage = 10;
            } else if (currentPage + 4 >= totalPages) {
                startPage = totalPages - 9;
                endPage = totalPages;
            } else {
                startPage = currentPage - 5;
                endPage = currentPage + 4;
            }
        }

        // calculate start and end item indexes
        var startIndex = (currentPage - 1) * pageSize;
        var endIndex = Math.min(startIndex + pageSize - 1, totalItems - 1);

        // create an array of pages to ng-repeat in the pager control
        // var pages = Array.from({length: end - start}, (v, k) => k + start)
        var pages = Array.from({ length: endPage + 1 - startPage }, (v, k) => k + startPage);

        // return object with all pager properties required by the view
        return {
            totalItems: totalItems,
            currentPage: currentPage,
            totalPages: totalPages,
            startPage: startPage,
            endPage: endPage,
            startIndex: startIndex,
            endIndex: endIndex,
            pages: pages
        };
    }

    render() {
        const pager = this.state.pager;

        if (!pager.pages || pager.pages.length <= 1) {
            return null;
        }

        return (
            <Wrapper>
                <PageLink disabled={pager.currentPage === 1} onClick={() => this.setPage(1)}>
                    &laquo;
                    </PageLink>
                <PageLink disabled={pager.currentPage === 1} onClick={() => this.setPage(pager.currentPage - 1)}>
                    &#8592;
                    </PageLink>
                {pager.pages.map((page, index) =>
                    <PageLink key={index}
                        active={pager.currentPage === page}
                        onClick={() => this.setPage(page)}>{page}</PageLink>
                )}
                <PageLink disabled={pager.currentPage === pager.totalPages}
                    onClick={() => this.setPage(pager.currentPage + 1)}>
                    &#8594;
                </PageLink>
                <PageLink disabled={pager.currentPage === pager.totalPages}
                    onClick={() => this.setPage(pager.totalPages)}>
                    &raquo;
                </PageLink>
            </Wrapper>
        );

    }
}

GiphPagination.propTypes = propTypes;
GiphPagination.defaultProps = defaultProps;