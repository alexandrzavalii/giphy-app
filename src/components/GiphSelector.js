import React from 'react';
import PropTypes from 'prop-types';

export const GiphSelector = ({availableCategories, handleClick}) => {

    let categories = availableCategories.length > 0 ?
    availableCategories.map(category => 
    <button key={category} onClick={()=>handleClick(category)}>{category}</button>) 
    : null;
    return categories;
}

GiphSelector.propTypes = {
    availableCategories: PropTypes.array.isRequired,
    handleClick: PropTypes.func.isRequired
}