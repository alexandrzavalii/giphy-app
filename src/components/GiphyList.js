import React from 'react';
import PropTypes from 'prop-types';

export const GiphyList = ({data}) => {
    let gifs = data.length > 0 ?
      data.map((gif) => {
         return (
             <div className='gif' key={gif.id}>
               <img src={gif.images.fixed_width.url} alt={gif.slug}/>
             </div>
         );
       }) :
       <p> No Gifs found </p>;
       return <div className='gif-list' style={styles.gifList}>{gifs}</div>;
    };

  GiphyList.propTypes = {
      data: PropTypes.array
  }


  const styles = {
    gifList: {
      display: 'flex',
      flexWrap: 'wrap',
    }
  }