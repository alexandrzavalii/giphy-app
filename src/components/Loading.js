import React from 'react';
import styled from 'styled-components';

const Loader = styled.div`
    position: absolute;
    top: calc(50% - 4em);
    left: calc(50% - 4em);
    width: 6em;
    height: 6em;
    border: 1.1em solid rgba(0, 0, 0, 0.2);
    border-left: 1.1em solid ${props => props.theme.base };
    border-radius: 50%;
    animation: load8 1.1s infinite linear;
    @keyframes load8 {
            0% {
            transform: rotate(0deg);
            }
            100% {
            transform: rotate(360deg);
            }
        }
`;

export const Loading = () => <Loader />;