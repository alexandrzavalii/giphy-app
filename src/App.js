import React, { Component } from 'react';
import { getGiphys, abortGetGiphys } from './api.js';
import { GiphyList } from "./components/GiphyList";
import { CategorySelector } from "./components/CategorySelector";
import { Pagination } from "./components/Pagination";
import { Loading } from "./components/Loading";
import { debounced } from './utils';
import { AVAILABLE_CATEGORIES } from './settings';


class App extends Component {
  constructor() {
    super();
    this.state = {
      data: [],
      category: AVAILABLE_CATEGORIES[0],
      totalItems: 0,
      loading: true,
      selectedGiph: null
    }
    this.selectCategory = this.selectCategory.bind(this);
    this.selectPage = this.selectPage.bind(this);
    this.selectGiph = this.selectGiph.bind(this);
  }



  componentDidMount() {
    setTimeout(() => getGiphys(this.state.category).then(({ data, pagination }) => {
      this.setState((prevState, props) => ({
        data,
        totalItems: pagination.total_count,
        loading: false
      }));
    }), 1000)

  }

  handleGet(category, offset) {
    const setCategory = category;
    this.setState({ loading: true, category }, () => {
      abortGetGiphys();
      debounced(500,
        getGiphys(category, offset).then(({ data, pagination }) =>
          this.setState((prevState, props) => {
            const state = {};
            
            if (prevState.category !== setCategory) {
              state.category = setCategory;
            }
            if(prevState.totalItems !== pagination.total_count) {
              state.totalItems = pagination.total_count
            }
            return {
              data, loading: false, selectCategory: null, ...state
            }
          })
          // prevState.category !== setCategory ?
          //   ({ data, totalItems: pagination.total_count, loading: false, selectedGiph: null }) :
          //   ({ data, loading: false, selectedGiph: null }))
        ))
    });
  }


  selectCategory(category) {
    this.handleGet(category, 0);
  }

  selectPage(offset) {
    this.handleGet(this.state.category, offset);
  }
  
  selectGiph(selectedGiph) {
    this.setState({ selectedGiph })
  }
  render() {
    return (
      <div className="App">
        <CategorySelector 
          activeCategory={this.state.category} 
          availableCategories={AVAILABLE_CATEGORIES} 
          handleClick={this.selectCategory} />
        {this.state.loading && <Loading />}
        <Pagination
          handlePageClick={this.selectPage}
          totalItems={this.state.totalItems}
        />
        {!this.state.loading &&
          <GiphyList
            selectedGiph={this.state.selectedGiph}
            handleGiphInteraction={this.selectGiph}
            data={this.state.data} />
        }

      </div>
    );
  }
}

export default App;
