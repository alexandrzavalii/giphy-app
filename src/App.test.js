import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Enzyme, {shallow} from 'enzyme';

import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import { mockResponse } from '../mocks';
// import components
import {GiphSelector} from './components/GiphSelector';

import { GiphyList } from './components/GiphyList';
// import { getGiphys } from './api';



it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});


// describe('<App />', () => {
//   it('should render correctly with data', () => {
//     const tree = renderer.create(<GiphyList data={mockResponse.data} handleGiphInteraction={()=>null}/>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });


describe('<GiphyList />', () => {
  it('should render correctly with data', () => {
    const tree = renderer.create(<GiphyList data={mockResponse.data} handleGiphInteraction={() => null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});


test('Select Dogs category', () => {
  // Render a checkbox with label in the document

  const Wrapper = mount(<GiphSelector activeCategory="Cats" availableCategories={['Cats','Dogs']} handleClick={()=>null}/>);
  const DogsButton = Wrapper.instance()
  // expect(secondButton.props().disabled).toEqual(true);

  // expect(checkbox.text()).toEqual('Off');

  // checkbox.find('input').simulate('change');

  // expect(checkbox.text()).toEqual('On');
});


//   it('gets search results', () => {
//     const response = mockTYM;
//     axios.get = jest.fn(() => Promise.resolve(response));

//     let wrapper = mount(<Giphy />);

//     wrapper.instance().searchGiphs('toro y moi', 5);
//     wrapper.instance().searchGiphs('toro y moi');


//   });

//   it('get next page of results', () => {
//     // const response = mockTYM;
//     // axios.get = jest.fn(() => Promise.resolve(response));

//     let wrapper = mount(<Giphy />);
//     wrapper.instance().state.data = mockTYM.data
//     wrapper.instance().state.total_count = mockTYM.pagination.total_count
//     console.log(wrapper.instance().state.total_count);
//     wrapper.instance().state.term = 'toro y moi';
//     let searchBtn = wrapper.find('button.search-btn');
//     searchBtn.simulate('click');

//     const nextBtn = wrapper.find('button.next');

//     nextBtn.simulate('click');


//   });

// });
