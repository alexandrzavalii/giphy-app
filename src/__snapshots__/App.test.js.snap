// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly with data 1`] = `
<section
  className="sc-bwzfXH fYCLEG"
>
  <img
    alt="cats-light-sabers-Ov5NiLVXT8JEc"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/Ov5NiLVXT8JEc/200w.gif"
  />
  <img
    alt="cats-blanket-W3QKEujo8vztC"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/W3QKEujo8vztC/200w.gif"
  />
  <img
    alt="cat-funny-animation-aC45M5Q4D07Pq"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/aC45M5Q4D07Pq/200w.gif"
  />
  <img
    alt="cat-cats-OmK8lulOMQ9XO"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/OmK8lulOMQ9XO/200w.gif"
  />
  <img
    alt="aEXP6scfSSwQo"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/aEXP6scfSSwQo/200w.gif"
  />
  <img
    alt="cat-fail-JfLdIahamXQI0"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/JfLdIahamXQI0/200w.gif"
  />
  <img
    alt="cat-sweet-eyes-9JLQKmspQAMWQ"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/9JLQKmspQAMWQ/200w.gif"
  />
  <img
    alt="producthunt-cats-music-streaming-3o72EX5QZ9N9d51dqo"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/3o72EX5QZ9N9d51dqo/200w.gif"
  />
  <img
    alt="cat-lol-peeks-jS6sVMK2fu4Uw"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/jS6sVMK2fu4Uw/200w.gif"
  />
  <img
    alt="kitty-cats-funny-12ipPASHq1luj6"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media2.giphy.com/media/12ipPASHq1luj6/200w.gif"
  />
  <img
    alt="afv-funny-lol-cats-3o85xkpCruOJPGtBMQ"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/3o85xkpCruOJPGtBMQ/200w.gif"
  />
  <img
    alt="afvpets-cats-pets-afv-3oEduUVL7wX5a1NBXq"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/3oEduUVL7wX5a1NBXq/200w.gif"
  />
  <img
    alt="cats-headwear-transmissions-iOmFvna43V78c"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/iOmFvna43V78c/200w.gif"
  />
  <img
    alt="cat-talking-ucEr1xRMsj6Fi"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/ucEr1xRMsj6Fi/200w.gif"
  />
  <img
    alt="cats-gl8ymnpv4Sqha"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media2.giphy.com/media/gl8ymnpv4Sqha/200w.gif"
  />
  <img
    alt="cat-animals-being-jerks-dicks-NqZn5kPN8VVrW"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/NqZn5kPN8VVrW/200w.gif"
  />
  <img
    alt="aww-edition-ExMGjbktr4phe"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/ExMGjbktr4phe/200w.gif"
  />
  <img
    alt="cheezburger-cats-glasses-nerds-xT77XZrTKOxycjaYvK"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media2.giphy.com/media/xT77XZrTKOxycjaYvK/200w.gif"
  />
  <img
    alt="na-gon-then-oMV9WErqn3ZG8"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/oMV9WErqn3ZG8/200w.gif"
  />
  <img
    alt="cats-10rW4Xw9eO0RmU"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/10rW4Xw9eO0RmU/200w.gif"
  />
  <img
    alt="cat-star-wars-9gISqB3tncMmY"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/9gISqB3tncMmY/200w.gif"
  />
  <img
    alt="cats-spin-cat-gif-11c7UUfN4eoHF6"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/11c7UUfN4eoHF6/200w.gif"
  />
  <img
    alt="request-cats-5oBRpcOT04dq0"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media2.giphy.com/media/5oBRpcOT04dq0/200w.gif"
  />
  <img
    alt="cute-pig-cat-hKrpeqRJowwi4"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/hKrpeqRJowwi4/200w.gif"
  />
  <img
    alt="day-cake-W7dBXzbnEpOBG"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/W7dBXzbnEpOBG/200w.gif"
  />
</section>
`;

exports[`<GiphyList /> should render correctly with data 1`] = `
<section
  className="sc-bwzfXH fYCLEG"
>
  <img
    alt="cats-light-sabers-Ov5NiLVXT8JEc"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/Ov5NiLVXT8JEc/200w.gif"
  />
  <img
    alt="cats-blanket-W3QKEujo8vztC"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/W3QKEujo8vztC/200w.gif"
  />
  <img
    alt="cat-funny-animation-aC45M5Q4D07Pq"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/aC45M5Q4D07Pq/200w.gif"
  />
  <img
    alt="cat-cats-OmK8lulOMQ9XO"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/OmK8lulOMQ9XO/200w.gif"
  />
  <img
    alt="aEXP6scfSSwQo"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/aEXP6scfSSwQo/200w.gif"
  />
  <img
    alt="cat-fail-JfLdIahamXQI0"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/JfLdIahamXQI0/200w.gif"
  />
  <img
    alt="cat-sweet-eyes-9JLQKmspQAMWQ"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/9JLQKmspQAMWQ/200w.gif"
  />
  <img
    alt="producthunt-cats-music-streaming-3o72EX5QZ9N9d51dqo"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/3o72EX5QZ9N9d51dqo/200w.gif"
  />
  <img
    alt="cat-lol-peeks-jS6sVMK2fu4Uw"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/jS6sVMK2fu4Uw/200w.gif"
  />
  <img
    alt="kitty-cats-funny-12ipPASHq1luj6"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media2.giphy.com/media/12ipPASHq1luj6/200w.gif"
  />
  <img
    alt="afv-funny-lol-cats-3o85xkpCruOJPGtBMQ"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/3o85xkpCruOJPGtBMQ/200w.gif"
  />
  <img
    alt="afvpets-cats-pets-afv-3oEduUVL7wX5a1NBXq"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/3oEduUVL7wX5a1NBXq/200w.gif"
  />
  <img
    alt="cats-headwear-transmissions-iOmFvna43V78c"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/iOmFvna43V78c/200w.gif"
  />
  <img
    alt="cat-talking-ucEr1xRMsj6Fi"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/ucEr1xRMsj6Fi/200w.gif"
  />
  <img
    alt="cats-gl8ymnpv4Sqha"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media2.giphy.com/media/gl8ymnpv4Sqha/200w.gif"
  />
  <img
    alt="cat-animals-being-jerks-dicks-NqZn5kPN8VVrW"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/NqZn5kPN8VVrW/200w.gif"
  />
  <img
    alt="aww-edition-ExMGjbktr4phe"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/ExMGjbktr4phe/200w.gif"
  />
  <img
    alt="cheezburger-cats-glasses-nerds-xT77XZrTKOxycjaYvK"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media2.giphy.com/media/xT77XZrTKOxycjaYvK/200w.gif"
  />
  <img
    alt="na-gon-then-oMV9WErqn3ZG8"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/oMV9WErqn3ZG8/200w.gif"
  />
  <img
    alt="cats-10rW4Xw9eO0RmU"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/10rW4Xw9eO0RmU/200w.gif"
  />
  <img
    alt="cat-star-wars-9gISqB3tncMmY"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/9gISqB3tncMmY/200w.gif"
  />
  <img
    alt="cats-spin-cat-gif-11c7UUfN4eoHF6"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media1.giphy.com/media/11c7UUfN4eoHF6/200w.gif"
  />
  <img
    alt="request-cats-5oBRpcOT04dq0"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media2.giphy.com/media/5oBRpcOT04dq0/200w.gif"
  />
  <img
    alt="cute-pig-cat-hKrpeqRJowwi4"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media3.giphy.com/media/hKrpeqRJowwi4/200w.gif"
  />
  <img
    alt="day-cake-W7dBXzbnEpOBG"
    className="sc-bdVaJa LLYPZ"
    onMouseOver={[Function]}
    selected={false}
    src="https://media0.giphy.com/media/W7dBXzbnEpOBG/200w.gif"
  />
</section>
`;

exports[`<GiphyList /> should render correctly with no data 1`] = `
<section
  className="sc-bwzfXH fYCLEG"
>
  <p>
     No Gifs found 
  </p>
</section>
`;
